### PR TITLE
[BEAM-6144] Add support for the autoscaling flags to the GO SDK.

### PR DESCRIPTION
**Pull request by bramp**
Add support for the autoscaling flags to the GO SDK.

--autoscaling_algorithm
--max_num_workers

Inline with what's documented here: https://cloud.google.com/dataflow/docs/guides/specifying-exec-params

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: sdks/go/pkg/beam/runners/dataflow/dataflowlib/job.go [hunk 1]
```diff
@@ -140,6 +144,18 @@ func Translate(p *pb.Pipeline, opts *JobOptions, workerURL, jarURL, modelURL str
...
 		job.Environment.WorkerPools[0].NumWorkers = opts.NumWorkers
 	}
+	if opts.Algorithm != "" {
+		settings := &df.AutoscalingSettings{
+			Algorithm: map[string]string{
+				"NONE":             "AUTOSCALING_ALGORITHM_NONE",
+				"THROUGHPUT_BASED": "AUTOSCALING_ALGORITHM_BASIC",
+			}[opts.Algorithm],
+		}
+		if opts.MaxNumWorkers > 0 {
```
Comment (aaltay) [inline sdks/go/pkg/beam/runners/dataflow/dataflowlib/job.go]: `MaxNumWorkers` could be set even if `opts.Algorithm` is not set. If Algorithm is not set, Dataflow will choose the appropriate autoscaling algortihm and max num workers would still be used for that.

**Comment by bramp** [comment]
Please take another look.

**Comment by lostluck** [comment]
I didn't see this until today. Thanks for the contribution!
LGTM.

R: @aaltay Please merge.

**Comment by lostluck** [comment]
R: @aaltay I typo'd a function in a previous commit, breaking top.  I've filed https://issues.apache.org/jira/browse/BEAM-6682 to improve this.

**Pull request merged**